atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/DefaultSingleBucketSweepTaskTest.java

```diff
@@ -0,0 +1,433 @@
+        when(boundRetriever.getStrictUpperBoundForCompletelyClosedBuckets()).thenReturn(context.bucketIdentifier());
+
+        assertThat(defaultSingleBucketSweepTask.runOneIteration(context.sweepableBucket()))
+                .isEqualTo(0);
+        verifyNoInteractions(completionListener, sweepQueueReader, sweepQueueDeleter, sweepQueueCleaner);
+    }
+
+    @ParameterizedTest
+    @MethodSource("allSweepBuckets")
+    public void returnsEarlyIfSweepTimestampIsBehindPastBucketProgress(SweepBucketTestContext context) {
+        when(sweepTimestampSupplier.getAsLong()).thenReturn(context.startTimestampInclusive() + 4L);
+        when(bucketProgressStore.getBucketProgress(context.bucket()))
+                .thenReturn(Optional.of(BucketProgress.createForTimestampProgress(5L)));
+
+        assertThat(defaultSingleBucketSweepTask.runOneIteration(context.sweepableBucket()))
+                .isEqualTo(0);
+        verifyNoInteractions(completionListener, sweepQueueReader, sweepQueueDeleter, sweepQueueCleaner);
+    }
+
+    @ParameterizedTest
+    @MethodSource("allSweepBuckets")
+    public void returnsEarlyIfSweepTimestampIsEqualToPastBucketProgress(SweepBucketTestContext context) {
+        when(sweepTimestampSupplier.getAsLong()).thenReturn(context.startTimestampInclusive() + 7L);
+        when(bucketProgressStore.getBucketProgress(context.bucket()))
+                .thenReturn(Optional.of(BucketProgress.createForTimestampProgress(7L)));
+
+        assertThat(defaultSingleBucketSweepTask.runOneIteration(context.sweepableBucket()))
+                .isEqualTo(0);
+        verifyNoInteractions(completionListener, sweepQueueReader, sweepQueueDeleter, sweepQueueCleaner);
+    }
+
+    @ParameterizedTest
+    @MethodSource("closedSweepBuckets")
+    public void sweepsEmptyDeleteableBucketEntirelyAndDeletesBucket(SweepBucketTestContext context) {
+        when(sweepTimestampSupplier.getAsLong()).thenReturn(context.endTimestampExclusive());
+        when(sweepQueueReader.getNextBatchToSweep(
+                        context.shardAndStrategy(),
+                        context.startTimestampInclusive() - 1,
+                        context.endTimestampExclusive()))
+                .thenReturn(SweepBatchWithPartitionInfo.of(
+                        SweepBatch.of(
+                                ImmutableSet.of(),
+                                DedicatedRows.of(ImmutableList.of()),
+                                context.endTimestampExclusive() - 1),
+                        ImmutableSet.of()));
+        when(boundRetriever.getStrictUpperBoundForCompletelyClosedBuckets()).thenReturn(Long.MAX_VALUE);
+
+        defaultSingleBucketSweepTask.runOneIteration(context.sweepableBucket());
+        verify(sweepQueueDeleter).sweep(eq(ImmutableList.of()), eq(Sweeper.of(context.shardAndStrategy())));
+        verify(sweepQueueCleaner)
+                .clean(
+                        eq(context.shardAndStrategy()),
```

Comment:
nits: You don't need eq() around the concrete objects if you're applying it to all arguments! Was there a readability preference for including it?